crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1136,24 +1138,67 @@ pub struct RpcServerConfig {
     ipc_endpoint: Option<String>,
     /// JWT secret for authentication
     jwt_secret: Option<JwtSecret>,
+    /// Configurable RPC middleware
+    rpc_middleware: RpcServiceBuilder<RpcMiddleware>,
 }
 
 // === impl RpcServerConfig ===
 
-impl RpcServerConfig {
+impl Default for RpcServerConfig<Identity> {
+    /// Create a new config instance
+    fn default() -> Self {
+        Self {
+            http_server_config: None,
+            http_cors_domains: None,
+            http_addr: None,
+            ws_server_config: None,
+            ws_cors_domains: None,
+            ws_addr: None,
+            ipc_server_config: None,
+            ipc_endpoint: None,
+            jwt_secret: None,
+            rpc_middleware: RpcServiceBuilder::new(),
+        }
+    }
+}
+
+use jsonrpsee::server::middleware::rpc::{RpcService, RpcServiceT};
+use tower::Layer;
+
+impl<RpcMiddleware> RpcServerConfig<RpcMiddleware>
+where
+    RpcMiddleware: for<'a> Layer<RpcService, Service: RpcServiceT<'a>> + Clone + Send + 'static,
+    <RpcMiddleware as Layer<RpcService>>::Service: Send + std::marker::Sync,
+{
```

Comment:
We need several implant blocks for this type, and we want to be as unrestrictive as possible, so the setters are easy to call

we only need to enforce these bounds when we start the server